src/release/release.ts

```
@@ -487,28 +504,77 @@ export class Release extends Component {
   /**
    * @returns a workflow or `undefined` if github integration is disabled.
    */
+  // TODO: did this method get too long now? What's the standard in this project?
   private createWorkflow(
     branchName: string,
     branch: Partial<BranchOptions>
   ): TaskWorkflow | undefined {
     const workflowName = branch.workflowName ?? `release-${branchName}`;
 
+    const sharedEnv: Record<string, string> = {};
+    if (branch.majorVersion !== undefined) {
+      sharedEnv.MAJOR = branch.majorVersion.toString();
+    }
+    if (branch.prerelease) {
+      sharedEnv.PRERELEASE = branch.prerelease;
+    }
+
+    if (branch.tagPrefix) {
+      sharedEnv.RELEASE_TAG_PREFIX = branch.tagPrefix;
+    }
+
+    const checkReleaseEnv: Record<string, string> = { ...sharedEnv };
+    if (
+      branch.skipConventionalCommitScopes &&
+      branch.skipConventionalCommitScopes.length > 0
+    ) {
+      checkReleaseEnv.SKIPPED_SCOPES =
+        branch.skipConventionalCommitScopes.join(",");
+    }
+    if (
+      branch.skipConventionalCommitTypes &&
+      branch.skipConventionalCommitTypes.length > 0
+    ) {
+      checkReleaseEnv.SKIPPED_TYPES =
+        branch.skipConventionalCommitTypes.join(",");
+    }
+
+    // if this is the release for "main" or "master", just call it "release:check".
+    // otherwise, "release:check:BRANCH"
+    // TODO: this will conflict if the user configures a branch called "check"
```

Is this important to handle?
The existing check for `main` vs `master` also fails if users happen to use both 😄

---

Yes, I think we might have to call it `check-release` or `release-check` or `pre-release` (<-- I like this one) if we go with a pre branch model.

---

Side bar: I don't like how this is currenlty done. It shouldn't check for a hardcoded branch name, but rather give the default branch the "nice" name.

---

Re `pre-release`:
I feel that `release-check` or `check-release` makes the current implementation a bit more obvious. I wouldn't expect that a failing `pre-release` task silently skips everything. Also, the `postBuildSteps` could be interpreted as pre release steps, too (if one interprets releasing = publishing).